packages/helpers/source/bignumber.test.ts

```diff
@@ -2,14 +2,58 @@ import { describe } from "@payvo/sdk-test";
 
 import { BigNumber } from "./bignumber";
 
-describe("BigNumber", async ({ assert, beforeEach, it, nock, loader }) => {
+describe("BigNumber", async ({ assert, beforeEach, it }) => {
 	beforeEach((context) => (context.subject = BigNumber.make(1)));
 
-	it("#decimalPlaces", () => {
+	it("#toString should succeed when input is provided as string", () => {
+		assert.is(BigNumber.make("0").toString(), "0");
+		assert.is(BigNumber.make("1").toString(), "1");
+		assert.is(BigNumber.make("1.5").toString(), "1.5");
+		assert.is(BigNumber.make("1.500000000001").toString(), "1.500000000001");
+	});
+
+	it("#toString should succeed when input is provided as number", () => {
+		assert.is(BigNumber.make(0).toString(), "0");
+		assert.is(BigNumber.make(1).toString(), "1");
+		assert.is(BigNumber.make(1.5).toString(), "1.5");
+		assert.is(BigNumber.make(1.500_000_000_001).toString(), "1.500000000001");
+	});
+
+	it("#toString should succeed when input is provided as bigint", () => {
+		assert.is(BigNumber.make(BigInt(0)).toString(), "0");
+		assert.is(BigNumber.make(BigInt(1)).toString(), "1");
+		assert.is(BigNumber.make(BigInt(10)).toString(), "10");
+		assert.is(BigNumber.make(BigInt(1_234_567_890)).toString(), "1234567890");
+	});
+
+	it("#toString should succeed when input is provided as BigNumber", () => {
+		assert.is(BigNumber.make(BigNumber.ONE).toString(), "1");
+		assert.is(BigNumber.make(BigNumber.make(1.5)).toString(), "1.5");
+		assert.is(BigNumber.make(BigNumber.make(1.500_000_000_001)).toString(), "1.500000000001");
+		assert.is(BigNumber.make(BigNumber.make("1.5")).toString(), "1.5");
+		assert.is(BigNumber.make(BigNumber.make("1.500000000001")).toString(), "1.500000000001");
+	});
+
+	it("#toString should not return a value having more than the specified decimals", () => {
+		assert.is(BigNumber.make("1", 8).toString(), "1");
+		assert.is(BigNumber.make(1, 8).toString(), "1");
+
+		assert.is(BigNumber.make("1.5", 8).toString(), "1.5");
+		assert.is(BigNumber.make("1.500000000000", 8).toString(), "1.5");
+		assert.is(BigNumber.make(1.5, 8).toString(), "1.5");
+
+		assert.is(BigNumber.make("1.500000005555", 8).toString(), "1.5");
+		assert.is(BigNumber.make(1.500_000_005_555, 8).toString(), "1.5");
+
+		assert.is(BigNumber.make("1.500000015555", 8).toString(), "1.50000001");
+		assert.is(BigNumber.make(1.500_000_015_555, 8).toString(), "1.50000001");
+	});
+
+	it("#decimalPlaces should succeed", () => {
 		assert.is(BigNumber.make("12.3456789").decimalPlaces(0).valueOf(), "12");
-		assert.is(BigNumber.make("12.3456789").decimalPlaces(2).valueOf(), "12.35");
-		assert.is(BigNumber.make("12.3456789").decimalPlaces(4).valueOf(), "12.3457");
-		assert.is(BigNumber.make("12.3456789").decimalPlaces(6).valueOf(), "12.345679");
+		assert.is(BigNumber.make("12.3456789").decimalPlaces(2).valueOf(), "12.34");
```

Comment:
@faustbrian is it okay if we don't round here?

Reply:
Probably best if we don't modify the value. Will see if we run into issues without rounding.